hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -311,7 +311,9 @@ private List<PartitionInfo> fetchPartitionInfos(KafkaConsumer consumer, String t
     do {
       partitionInfos = consumer.partitionsFor(topicName);
       try {
-        TimeUnit.SECONDS.sleep(10);
+        if (partitionInfos == null) {
+          TimeUnit.SECONDS.sleep(10);
```

Comment:
We should get rid of sleeping altogether by using proper retrying client

Reply:
@alexeykudinkin do you have an idea which a proper retrying client we can use here? It is probably not necessary to make it as a part of this PR, due to the triviality of this change. However, I agree that this issue should be solved at the root of the problem.

Reply:
@vburenin we can implement one ourselves, it's fairly straightforward flow

Reply:
I created a Jira ticket for this improvement as a follow-up: HUDI-4838.  Let's land this PR as is.